### ISSUE TITLE
How wayfire manage multiple monitor and fullscreen application

### ISSUE THREAD

**Issue by wisperC4** [WayfireWM/wayfire#1846]
Hello, 
I'm using multiple monitor configuration, when i start a fullscreen application, which monitor is chosen and can i modify it ? 
At this time it's seem some magic or random
Regards,
Nicolas

**Comment by marcusbritanicus** [comment]
If I'm not totally mistaken, Wayfire will open a new window or full-screen a window on the current screen. If you switch to the screen you want (move your mouse there and click) and then start your app, then it must open the app in that screen.

**Comment by ammen99** [comment]
Apps may also request to be fullscreened on a specific output, which wayfire also tries to take into account.

**Comment by wisperC4** [comment]
Thanks for the explanation, but how can I do ?
For example, I'm using virt-viewer ( a spice client for qemu) with a
fullscreen option.
Virt-viewer detect the two monitor with correct resolution, create 2
windows,  but wayfire put one windows by monitor but randomly ( in my point
of view )
NIcolas

Le mar. 1 août 2023 à 08:27, Ilia Bozhinov ***@***.***> a
écrit :

**Comment by marcusbritanicus** [comment]
Firstly, I think virt-viewer X11 based and does not currently support wayland. Which would mean that virt-viewer would be running via Xwayland on wayfire. In such a case I think, it would not be possible for it to place the correct window on the correct screen.

Secondly, if the two windows have distinct titles, you may be able to use window rules to place your fullscreen views.

PS: These are my guesses and I may be wrong on both counts.

**Comment by wisperC4** [comment]
virt-viewer is based gtk and seems to be compatible with wayland when
WAYLAND_DISPLAY  & XDG env is detected.


Le mar. 1 août 2023 à 09:30, Marcus Britanicus ***@***.***> a
écrit :

**Comment by marcusbritanicus** [comment]
@wisperC4 You may find this interesting: https://stackoverflow.com/questions/39367246/select-certain-monitor-for-going-fullscreen-with-gtk

GTK (at least GTK3) does not really support full-screening windows on particular screens. The fix mentioned there would work for X11 but not for Wayland - clients are not allowed to place themselves in Wayland. For that matter, even Qt cannot place full-screen windows on Wayland.

Looks like you can try using window-rules, although I do not know if it will help you much. If you know python, you can write a python script to do this via IPC.

**Comment by ammen99** [comment]
TLDR of the discussion here: If the applications have different app-id's or titles, then you can use window-rules (or a script based on ipc-rules) to match them to the appropriate outputs, otherwise the apps themselves have to request the correct output in the fullscreen request (which doesn't seem possible in gtk3).
